applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/mechanical_explicit_strategy.hpp

```diff
@@ -182,20 +182,20 @@ class MechanicalExplicitStrategy
     // initial operations ... things that are constant over the Solution Step
     pScheme->InitializeSolutionStep(r_model_part, matrix_a_dummy, mDx, mb);
 
-    ProcessInfo &r_current_process_info = r_model_part.GetProcessInfo();
-    ElementsArrayType &r_elements = r_model_part.Elements();
-
     if (BaseType::mRebuildLevel > 0) {
-    auto it_elem = r_model_part.ElementsBegin();
-// #pragma omp parallel for firstprivate(it_elem)
-    for (int i = 0; i < static_cast<int>(r_elements.size()); ++i) {
-        // Getting nodal mass and inertia from element
-        Vector dummy_vector;
-        // this function needs to be implemented in the respective
-        // element to provide inertias and nodal masses
-        (it_elem + i)
-            ->AddExplicitContribution(dummy_vector, RESIDUAL_VECTOR,
-                                    NODAL_INERTIA, r_current_process_info);
+      ProcessInfo &r_current_process_info = r_model_part.GetProcessInfo();
+      ElementsArrayType &r_elements = r_model_part.Elements();
+
+      auto it_elem = r_model_part.ElementsBegin();
+  // #pragma omp parallel for firstprivate(it_elem)
```

Comment:
what is the point of being explicit without OMP?

Reply:
I remember we had some problems here a while ago

Reply:
yes there are still some problems that we have not addressed yet, this is one of them

Reply:
do you want me to try to fix it?

Reply:
if you find a solution this would be nice :metal:

Reply:
let me finish something

Reply:
I think is fixed, do we have tests for this?

Reply:
can you push? I will test it

Reply:
It is fixed, I have done a test, but there ia problem with the check in the CL of the trust, one second

Reply:
BTW, the error wasn't a OMP error, was a memory leak

Reply:
I cleaned the code :D